test/unittest/tests/test_qsort.cpp

```diff
@@ -0,0 +1,171 @@
+/*-
+ * Copyright (c) 2014-present MongoDB, Inc.
+ * Copyright (c) 2008-2014 WiredTiger, Inc.
+ *	All rights reserved.
+ *
+ * See the file LICENSE for redistribution information.
+ */
+
+#include <algorithm>
+#include <vector>
+
+#include <catch2/catch.hpp>
+
+#include "wt_internal.h"
+
+namespace {
+
+const uint8_t COMPOUND_MAGIC_B = 123;
+const uint64_t COMPOUND_MAGIC_C = 0xdeadbeefBAADF00D;
+struct compound_test_type {
+    int a;
+    uint8_t b;
+    uint64_t c;
+};
+
+class random_generator {
+    std::random_device rnd_device;
+    std::mt19937 mersenne_engine;
+    std::uniform_int_distribution<int> dist;
+
+public:
+    random_generator() = default;
+    random_generator(const random_generator &) = delete;
+    random_generator &operator=(const random_generator &) = delete;
+    ~random_generator() = default;
+
+    std::vector<int>
+    make_vector(size_t size)
+    {
+        std::vector<int> rv(size);
+        auto rnd = [this]() { return this->dist(this->mersenne_engine); };
+        std::generate(begin(rv), end(rv), rnd);
+        return rv;
+    }
+
+    std::vector<struct compound_test_type>
+    make_compound_vector(size_t size)
+    {
+        std::vector<struct compound_test_type> rv(size);
+        auto rnd = [this]() {
+            struct compound_test_type t;
+            t.a = this->dist(this->mersenne_engine);
+            t.b = COMPOUND_MAGIC_B;
+            t.c = COMPOUND_MAGIC_C;
```

Comment:
I wanted this to be a POD type to ensure no vtable pointers etc. Same goes for the `<` and `<=>` operators (the spaceship doesn't work for another reason though, since we compile our C++ as C++ 17 rather than C++ 20).

Reply:
Non-virtual functions do not inflate instance size (no vtable): so I don't understand the argument. Apologies I mistakenly thought we were using C++20.
Also:
```
bool operator<(const compound_test_type&, const compound_test_type&);
```

Reply:
You're right, I forgot that. Done in 514dde4, thanks for the advice.